tiledb/sm/filter/filter_pipeline.cc

```diff
@@ -440,13 +486,32 @@ Status FilterPipeline::run_reverse(
 
       // Final filter: output directly into the shared output buffer.
       bool last_filter = filter_idx == 0;
+      // Assume there are type conversions in the pipeline.
+      bool conversions = true;
       if (last_filter) {
         void* output_chunk_buffer =
             static_cast<char*>(tile->data()) + chunk_data.chunk_offsets_[i];
         RETURN_NOT_OK(output_data.set_fixed_allocation(
             output_chunk_buffer, chunk.unfiltered_data_size_));
         reader_stats->add_counter(
             "read_unfiltered_byte_num", chunk.unfiltered_data_size_);
+        // Restore tile datatype to it's initial schema value.
```

Comment:
I think the tile created by the reader should have the right type from the get go. My understanding is that the filter pipeline will use temporary buffers until we reach the last filter, at which point we'll copy to the final tile. Am I incorrect? I don't think we should change the tile type after it's been created.